src/passes/OptimizeInstructions.cpp

```diff
@@ -774,11 +799,8 @@ struct OptimizeInstructions
           inner->type = Type::i32;
           return replaceCurrent(inner);
         }
-      }
-      {
```

Comment:
Why these changes? I know the duplication of the variables is not strictly necessary, but I think it is good to have each pattern match here be completely independent.